### ISSUE TITLE
Hangs behind HTTP Proxy

### ISSUE THREAD

**Issue by Toilal** [vuejs/vue-cli#785]
### Version
3.0.0-alpha.8

### Steps to reproduce
Run vue create on a computer configured with a corporate HTTP Proxy

(`http_proxy` and `https_proxy` environment variable are defined, everything else is working properly)

### What is expected?
It should run as usual

### What is actually happening?
vue create command hangs forever with no prompt at all.

---
Tested on Windows and Ubuntu.

**Comment by yyx990803** [comment]
Did it work with any earlier versions?

It's a bit difficult for me to simulate what your actual environment, but it probably hangs here: https://github.com/vuejs/vue-cli/blob/dev/packages/%40vue/cli/lib/util/getVersions.js#L17

I don't quite know why though because `axios` does automatically handle `https_proxy` env variables.

**Comment by Toilal** [comment]
I'll try to debug this at work. That's the first time I try v3 here, and it was working properly with v2.

**Comment by ayush987goyal** [comment]
Getting the following error with `@vue/cli@3.0.0-alpha.9` for `vue create my-app`
``` 
ERROR  Error: write EPROTO 101057795:error:140770FC:SSL routines:SSL23_GET_SERVER_HELLO:unknown protocol:openssl\ssl\s23_clnt.c:827:

Error: write EPROTO 101057795:error:140770FC:SSL routines:SSL23_GET_SERVER_HELLO:unknown protocol:openssl\ssl\s23_clnt.c:827:

    at _errnoException (util.js:1024:11)
    at WriteWrap.afterWrite [as oncomplete] (net.js:867:14)
```

**Comment by marco-carvalho** [comment]
Getting this error with `@vue/cli@3.0.0-alpha.9` for `vue create randomapp`

> ERROR  Error: connect ETIMEDOUT <someIpWithPortHere>
> Error: connect ETIMEDOUT <someIpWithPortHere>
>     at Object._errnoException (util.js:1003:13)
>     at _exceptionWithHostPort (util.js:1024:20)
>     at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1194:14)

**Comment by SMoni** [comment]
Experienced a similar problem on Windows without the hanging but with a `ERROR: connect ECONNREFUSED`.

Could solve the error by manually set HTTP_PROXY and HTTPS_PROXY. Then it became a `Error: socket hang up`.

After changing the line 

https://github.com/vuejs/vue-cli/blob/dev/packages/%40vue/cli/lib/util/getVersions.js#L15

to `http://registry.npmjs.org` I still get the same error, but through the line 

https://github.com/vuejs/vue-cli/blob/dev/packages/%40vue/cli/lib/util/installDeps.js#L19

which could be solved by `npm config set registry http://registry.npmjs.org/`

I hope nobody mind my question: In https://github.com/vuejs/vue-cli/blob/dev/packages/%40vue/cli/lib/util/installDeps.js#L46 the user settings are respected. In https://github.com/vuejs/vue-cli/blob/dev/packages/%40vue/cli/lib/util/getVersions.js#L17 not. Is there a purpose to do that?

**Comment by Toilal** [comment]
Maybe it's related to this axios issue : https://github.com/axios/axios/issues/925

**Comment by yyx990803** [comment]
If this is an axios issue we can switch to a different HTTP client. Can anyone try it out? I don't have the network environment to confirm whether that would fix it.

**Comment by SMoni** [comment]
I'm going to give it a try tomorrow and givin' feedback.

**Comment by Toilal** [comment]
Maybe https://github.com/request/request ? It's stated in the axios issue that this use case works out of the box with request.

**Comment by wadetandy** [comment]
Have verified changing out request works in my environment.  Unless someone else is already working on this I will put together a PR.

**Comment by yyx990803** [comment]
Thanks @wadetandy !

Code changes looks good to me. Can someone else in this thread try out https://github.com/vuejs/vue-cli/pull/829 to see if that fixes it for you?

**Comment by SMoni** [comment]
Going to check this in the morning, but's looking good.

**Comment by Toilal** [comment]
Thanks, it's working fine now for me !

**Comment by SMoni** [comment]
Also thanks, it's working for me too. At least if I hadn't muddled up some settings for our proxy :/

**Comment by marco-carvalho** [comment]
i'm getting this error using `@vue/cli@3.0.0-beta.1` for `vue create randomApp`

``
ERROR  RequestError: Error: connect ETIMEDOUT 151.101.92.162:443
RequestError: Error: connect ETIMEDOUT 151.101.92.162:443
    at new RequestError (C:\Users\MLJ\AppData\Roaming\npm\node_modules\@vue\cli\node_modules\request-promise-core\lib\errors.js:14:15)
    at Request.plumbing.callback (C:\Users\MLJ\AppData\Roaming\npm\node_modules\@vue\cli\node_modules\request-promise-core\lib\plumbing.js:87:29)
    at Request.RP$callback [as _callback] (C:\Users\MLJ\AppData\Roaming\npm\node_modules\@vue\cli\node_modules\request-promise-core\lib\plumbing.js:46:31)
    at self.callback (C:\Users\MLJ\AppData\Roaming\npm\node_modules\@vue\cli\node_modules\request\request.js:186:22)
    at Request.emit (events.js:160:13)
    at Request.onRequestError (C:\Users\MLJ\AppData\Roaming\npm\node_modules\@vue\cli\node_modules\request\request.js:878:8)
    at ClientRequest.emit (events.js:160:13)
    at TLSSocket.socketErrorListener (_http_client.js:389:9)
    at TLSSocket.emit (events.js:160:13)
    at emitErrorNT (internal/streams/destroy.js:64:8)
``

**Comment by jnal** [comment]
This work around work for me..

I have to modified manually the request.js and indicate proxy as part of reqOpts. It doesn't seem to respect the environment variables that I have. 
```
const request = require('request-promise-native')
module.exports = {
  async get (uri) {
    const reqOpts = {
      method: 'GET',
      resolveWithFullResponse: true,
      json: true,
      uri,
	  proxy: 'http://{domain}%5cu{username}:{pass}@proxy.company.com:8080'
    }
    return request(reqOpts)
  }
}
```

**Comment by wadetandy** [comment]
@jnal What variables are you using to set proxy normally?

**Comment by jnal** [comment]
i have all the variables below set in .npmrc but that doesn't seem to be respected by the request.

proxy=http://{domain}%5cu{username}:{pass}@proxy.company.com:8080
https-proxy=http://{domain}%5cu{username}:{pass}@proxy.company.com:8080
http-proxy=http://{domain}%5cu{username}:{pass}@proxy.company.com:8080
HTTP_PROXY=http://{domain}%5cu{username}:{pass}@proxy.company.com:8080
HTTPS_PROXY=http://{domain}%5cu{username}:{pass}@proxy.company.com:8080
ignore-ssl=true
strict-ssl=false

some of the settings above might be invalid but I had them anyway just to make sure I am not missiing anything. also in environment variables
HTTP_PROXY=http://{domain}%5cu{username}:{pass}@proxy.company.com:8080
HTTPS_PROXY=http://{domain}%5cu{username}:{pass}@proxy.company.com:8080

also I'm using windows 10, node 8.9.4 and npm 5.6.0 if that helps

**Comment by wvandeneede** [comment]
I'm getting this error `@vue/cli@3.0.0-beta.6` for `vue create testApp`

```
 ERROR  RequestError: Error: connect ETIMEDOUT 151.101.16.162:443
RequestError: Error: connect ETIMEDOUT 151.101.16.162:443
    at new RequestError (C:\Users\infodev4\AppData\Roaming\npm\node_modules\@vue\cli\node_modules\request-promise-core\lib\errors.js:14:15)
    at Request.plumbing.callback (C:\Users\infodev4\AppData\Roaming\npm\node_modules\@vue\cli\node_modules\request-promise-core\lib\plumbing.js:87:29)
    at Request.RP$callback [as _callback] (C:\Users\infodev4\AppData\Roaming\npm\node_modules\@vue\cli\node_modules\request-promise-core\lib\plumbing.js:46:31)
    at self.callback (C:\Users\infodev4\AppData\Roaming\npm\node_modules\@vue\cli\node_modules\request\request.js:186:22)
    at emitOne (events.js:116:13)
    at Request.emit (events.js:211:7)
    at Request.onRequestError (C:\Users\infodev4\AppData\Roaming\npm\node_modules\@vue\cli\node_modules\request\request.js:878:8)
    at emitOne (events.js:116:13)
    at ClientRequest.emit (events.js:211:7)
    at TLSSocket.socketErrorListener (_http_client.js:387:9)
```

We are behind a proxy and we are using Proget internally to manage our npm packages.
I am running tests with npm  5.8.0-next.0, which allows for a no-proxy variable to be set in npm config.

Does vue-cli take these npm settings into account?  We can not use any environment variables as they will interfere with other tools.

The `npm config list` looks like this:

```
https-proxy = "http://proxyserver:proxyport/"
no-proxy = "progetserver"
proxy = "http://proxyserver:proxyport/"
registry = "http://progetserver:progetport/npm/"
```

**Comment by DrDanRyan** [comment]
Did this ever get fixed? I am on 3.0.0 beta 10 and I still can't use `vue create` behind my corporate proxy.

**Comment by wadetandy** [comment]
@DrDanRyan this issue was resolved for me and a few others when my pr #829 was merged.

**Comment by fizxmike** [comment]
@wadetandy , how do I get a release with you PR in it? Version numer? Or does dev branch feed NPM directly?

**Comment by wadetandy** [comment]
My fix was merged and released with 3.0.0-alpha.10, so should be part of latest beta/RC.

**Comment by wadetandy** [comment]
And I have confirmed as it is working for me with those releases. May be a related but separate bug affecting you.
